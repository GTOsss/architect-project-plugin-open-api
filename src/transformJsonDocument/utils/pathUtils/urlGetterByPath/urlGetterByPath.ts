import { methodNameByPath } from '../methodNameByPath';
import { urlGetterTemplateLiteralsByPath } from '../urlGetterTemplateLiteralsByPath';
import { genCode } from '#/templates';
import { urlGetterParams } from '#/transformJsonDocument/utils/pathUtils/urlGetterParams';
import { upperFirst } from 'lodash';

/**
 * Генератор кода "url getter" функции.
 * */
export const urlGetterByPath = (openApiPath: string, method: string) => {
  let name = methodNameByPath(openApiPath);
  name = `${method}${upperFirst(name)}`;
  const body = urlGetterTemplateLiteralsByPath(openApiPath);
  let params = urlGetterParams(openApiPath).filter(Boolean).join();
  const typeParams = 'Record<string, string | number>';
  params = params ? `{ ${params} }: ${typeParams}` : '';
  return genCode(`~name~ = (~params~) => ~body~`, { name, params, body });
};
