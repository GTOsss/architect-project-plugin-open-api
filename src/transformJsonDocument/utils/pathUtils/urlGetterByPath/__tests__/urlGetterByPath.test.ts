import { urlGetterByPath } from '../urlGetterByPath';

describe('urlGetterByPath', () => {
  it('should create url getter function for /users', () => {
    expect(urlGetterByPath('/users', 'get')).toMatchSnapshot();
  });

  it('should create url getter function for /users/{userId}', () => {
    expect(urlGetterByPath('/users/{userId}', 'get')).toMatchSnapshot();
  });

  it('should create url getter function for /users/{userId}/groups/{groupId}', () => {
    expect(urlGetterByPath('/users/{userId}/groups/{groupId}', 'get')).toMatchSnapshot();
  });
});
