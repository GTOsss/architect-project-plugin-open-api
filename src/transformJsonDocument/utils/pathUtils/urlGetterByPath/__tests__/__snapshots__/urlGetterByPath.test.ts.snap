// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`urlGetterByPath should create url getter function for /users 1`] = `"getUsers = () => \`/users\`"`;

exports[`urlGetterByPath should create url getter function for /users/{userId} 1`] = `"getUsersByUserId = ({ userId }: Record<string, string | number>) => \`/users/\${userId}\`"`;

exports[`urlGetterByPath should create url getter function for /users/{userId}/groups/{groupId} 1`] = `"getUsersByUserIdGroupsByGroupId = ({ userId,groupId }: Record<string, string | number>) => \`/users/\${userId}/groups/\${groupId}\`"`;
